engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala

```diff
@@ -329,26 +318,29 @@ class MaterializeWorkflowDescriptorActor(serviceRegistryActor: ActorRef,
   /**
     * Map a call to a backend name depending on the runtime attribute key
     */
-  private def assignBackendUsingRuntimeAttrs(call: WdlTaskCall): Option[String] = {
-    val runtimeAttributesMap = call.task.runtimeAttributes.attrs
+  private def assignBackendUsingRuntimeAttrs(call: TaskCallNode): Option[String] = {
+    val runtimeAttributesMap = call.callable.runtimeAttributes.attributes
     runtimeAttributesMap.get(RuntimeBackendKey) map { wdlExpr => evaluateBackendNameExpression(call.fullyQualifiedName, wdlExpr) }
   }
 
-  private def evaluateBackendNameExpression(callName: String, backendNameAsExp: WdlExpression): String = {
-    backendNameAsExp.evaluate(NoLookup, NoFunctions) match {
-      case Success(runtimeString: WdlString) => runtimeString.valueString
-      case Success(x: WdlValue) =>
+  private def evaluateBackendNameExpression(callName: String, backendNameAsExp: WomExpression): String = {
+    backendNameAsExp.evaluateValue(Map.empty, NoIoFunctionSet) match {
+      case Valid(runtimeString: WdlString) => runtimeString.valueString
+      case Valid(x: WdlValue) =>
         throw new Exception(s"Non-string values are not currently supported for backends! Cannot use backend '${x.valueString}' to backend to Call: $callName")
-      case Failure(error) =>
-        throw new Exception(s"Dynamic backends are not currently supported! Cannot assign backend '${backendNameAsExp.valueString}' for Call: $callName", error)
+      case Invalid(errors) =>
+        // TODO WOM: need access to a "source string" for WomExpressions
+        // TODO WOM: ErrorOrify this ?
```

Comment:
Absolutely. Especially since this is in the middle of a bunch of other `ErrorOr` chains!